### ISSUE TITLE
Unify the plotting of band structures and DOSs

### ISSUE THREAD

**Issue by elinscott** [epfl-theos/koopmans#119]
Currently we have ad-hoc code at the end of the Wannierise, DFPT, and shortly the DFT workflows where we generate band structure and DOS plots.

We should instead have a unique routine for generating these plots.

One potential solution add the following to the `Workflow` class:

```
   _bs_to_plot = [] # A list where we store the band structures that we want to plot
   
   def run(self) -> None:
       self._run()
       self._plot_results()
   
   @abstractmethod
   def _run(self) -> None:
       ...
       
   def _plot_results(self) -> None:
       if self.parameters.calculate_bands:
          self._plot_bands(self):
   
   def add_bandstructure_for_plotting(self, calc: Calculator, label: str, ...):
       self._bs_to_plot.append({'bs': calc.results['band structure'], 'label': label, ...})
   
   def _plot_bands(self) -> None:
       if len(self._bs_to_plot) == 0:
           return
       <code here that plots the band structures>
```
This means that if a workflow can select a bandstructure to be plotted using `self.add_bandstructure_for_plotting()` and the plot will be generated at the conclusion of `self.run()`

**Comment by elinscott** [comment]
This was partially resolved by #106 but we still need to check how consistently we are shifting the VBE to zero, because this is causing issues.

**Comment by elinscott** [comment]
After more consideration the solution above is not ideal because we often have multiple plots per workflow.  We can leave band structure plot regeneration to the `*.fig.pkl` files
